lib/active_merchant/billing/gateways/alelo.rb

```diff
@@ -0,0 +1,245 @@
+require 'jose'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class AleloGateway < Gateway
+      self.test_url = 'https://sandbox-api.alelo.com.br/alelo/sandbox/'
+      self.live_url = 'https://desenvolvedor.alelo.com.br'
+
+      self.supported_countries = ['BR']
+      self.default_currency = 'BRL'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'https://www.alelo.com.br'
+      self.display_name = 'Alelo'
+
+      def initialize(options = {})
+        requires!(options, :client_id, :client_secret)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        post = {}
+        add_order(post, options)
+        add_amount(post, money)
+        add_payment(post, payment, options)
+        add_geolocation(post, options)
+        add_extra_data(post, options)
+
+        commit('capture/transaction', post, options)
+      end
+
+      def refund(money, authorization, options = {})
+        request_id = authorization.split('#').first
+        options[:http] = { method: :put, prevent_encrypt: true }
+        commit('capture/transaction/refund', { uuid: request_id }, options, :put)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        force_utf8(transcript.encode).
+          gsub(%r((Authorization: Bearer )[\w -]+), '\1[FILTERED]').
+          gsub(%r((client_id=|Client-Id:)[\w -]+), '\1[FILTERED]\2').
+          gsub(%r((client_secret=|Client-Secret:)[\w -]+), '\1[FILTERED]\2')
```

Comment:
We need `cardNumber` and `securityCode`, right?

Reply:
I see these scrubbed values are headers and hence not encrypted, while the boody itself is. Makes sense 👍